clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -131,7 +134,7 @@ public void prepare() throws AuthenticationException, IOException {
         } catch (AuthenticationException e) {
             // Clients are notified of authentication exceptions to enable operations to be terminated
             // without retries. Other errors are handled as network exceptions in Selector.
-            state = new ChannelState(ChannelState.State.AUTHENTICATION_FAILED, e);
+            state = new ChannelState(ChannelState.State.AUTHENTICATION_FAILED, e, remoteAddress.toString());
```

Comment:
Need null-check since `remoteAddress` is null on the server-side since we are currently only setting for client connectons.

Reply:
:+1: